src/Microsoft.Azure.EventHubs.Processor/AzureStorageCheckpointLeaseManager.cs

```diff
@@ -226,27 +226,21 @@ public async Task<bool> DeleteLeaseStoreAsync()
             return retval;
         }
 
-        public async Task<Lease> GetLeaseAsync(string partitionId) // throws URISyntaxException, IOException, StorageException
+        public Task<Lease> GetLeaseAsync(string partitionId) // throws URISyntaxException, IOException, StorageException
         {
-            AzureBlobLease retval = null;
-
             CloudBlockBlob leaseBlob = GetBlockBlobReference(partitionId);
 
-            if (await leaseBlob.ExistsAsync(null, this.operationContext).ConfigureAwait(false))
-            {
-                retval = await DownloadLeaseAsync(partitionId, leaseBlob).ConfigureAwait(false);
-            }
-
-            return retval;
+            return DownloadLeaseAsync(partitionId, leaseBlob);
         }
 
         public IEnumerable<Task<Lease>> GetAllLeases()
         {
-            IEnumerable<string> partitionIds = this.host.PartitionManager.GetPartitionIdsAsync().WaitAndUnwrapException();
+            // ListBlobsSegmentedAsync returns first 5000 blobs so no need to check for pagination.
+            var leaseBlobsResult = this.consumerGroupDirectory.ListBlobsSegmentedAsync(null).WaitAndUnwrapException();
```

Comment:
ListBlobsSegmentedAsync API may not guarantee that it will return all blobs (up to 5,000) in the first API call and hence we need to use BlobContinuationToken class to make sure all blobs are retrieved by checking the BlobContinuationToken object.